pages/login.tsx

```diff
@@ -1,21 +1,79 @@
 import { GetServerSidePropsContext } from "next"
-import {getSession, signIn} from "next-auth/react"
+import { getSession, signIn, signOut } from "next-auth/react"
 import HamburgerMenu from "../components/HamburgerMenu"
+import mongoose from "mongoose"
+import { UserModel } from "../models/user"
+import { useEffect } from "react"
+import { useRouter } from 'next/router'
 
-export default function Login(){
+
+
+export default function Login(props: {
+    failedAttempt: boolean,
+}) {
+
+    const router = useRouter()   //use this later to decide whether to show 'failed' message
+
+    useEffect(() => {
+        if (props.failedAttempt) {       //sign out user if log in unsucessful
+            signOut()      //this will refresh the page, run getServerSideProps again, returning { props: {failedAttempt:false} } so the useEffect hook doesnt run again
+        }
+    }, [])
+    //ok some weird glitch happened so i dont know if this method is any good
+    
 
     return (<>
-    <HamburgerMenu home aboutMe leaveReview blue/> 
-    <div className="container h-screen text-center">
-        <div className="top-1/2 relative translate-y-[-50%]">
-    <button onClick={()=>signIn("google")} className="bg-sky-700 text-white rounded-lg p-2  drop-shadow-lg">Sign in</button>
-    </div></div>
+        <HamburgerMenu home aboutMe leaveReview blue />
+
+        <div className="container h-screen text-center">
+            <div className="top-1/2 relative translate-y-[-50%]">
+                <button onClick={() => signIn("google")} className="bg-sky-700 text-white rounded-lg p-2 px-3 drop-shadow-lg">Continue with Google</button>
+
+                {(router.query.unauthorised === 'true' ?? false)
+                    && <div className="mt-16 "><span className="text-[16px] text-slate-700 bg-red-200 p-2 px-4 rounded-full">Your login was unauthorised</span></div>}
+            </div>
+
+        </div>
+
     </>)
-    
+
 }
 
-export async function getServerSideProps(context:GetServerSidePropsContext){
+export async function getServerSideProps(context: GetServerSidePropsContext) {
+
     const session = await getSession(context)
-    if (!session) return {props:{}}         //go back to sign in page
-    return {redirect :{permanent: false,destination :"/admin"}}
+    console.log("Session", session)
+    if (!session) {
+        return { props: { failedAttempt: false } }         //go back to sign in page
+    }
+
+    try {
+
+        await mongoose.connect(process.env.MONGODB_URL as string)
+        const thisUser = await UserModel.findOne({ email: session.user.email })
+
+
+        if (!thisUser) {                  //user not found in database
```

Comment:
Are the authorized users manually added to the database by you?

Reply:
Yes, or I temporarily uncomment the code below this to allow a new user to be created in the database